extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php

```diff
@@ -12,45 +12,118 @@ function __construct( ) {
 	}
 
 	public function index() {
-		wfProfileIn( __METHOD__ );
 		$this->wg->Out->setPageTitle( wfMsg( 'phalanx-stats-title' ) );
 		$this->wg->Out->addBacklinkSubtitle( $this->phalanxTitle );
 
 		if ( !$this->userCanExecute( $this->wg->User ) ) {
-			wfProfileOut( __METHOD__ );
 			$this->displayRestrictionError();
-			return;
+			return false;
 		}
 
 		$par = $this->getPar();
+
+		// Look for paths with a wiki ID like /wiki/Special:PhalanxStats/wiki/125951
 		if ( strpos( $par, 'wiki' ) === 0 ) {
 			// show per-wiki stats
-			list ( , $wikiId ) = explode( "/", $par, 2 );
-			$this->blockWikia( $wikiId );
-		} else {
-			$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
-			if ( !empty( $blockId ) ) {
-				// show block stats
-				$this->blockStats( $blockId );
-			} else {
-				// show help page
-				$this->forward( 'PhalanxStatsSpecial', 'help' );
-			}
+			return $this->forward( 'PhalanxStatsSpecial', 'blockWiki' );
+		}
+
+		// Look for paths with a block ID like /wiki/Special:PhalanxStats/123456
+		$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
+		if ( empty( $blockId ) ) {
+			// show help page
+			return $this->forward( 'PhalanxStatsSpecial', 'help' );
+		}
+
+		// show block stats
+		return $this->blockStats( $blockId );
+	}
+
+	/**
+	 * Show a list of blocks for a specific wiki
+	 *
+	 * @return bool
+	 */
+	public function blockWiki() {
```

Comment:
I think we can live with the simple check:

```
if ( !$this->userCanExecute( $this->wg->User ) ) {↵
    $this->displayRestrictionError();↵
    return false;↵
}↵
```
With current (lack of) implementation ACLs, a) we still need to call `userCanExecute()` whenever we want to limit execution; b) we still need to `displayRestrictionError()` in `$this` context; c) we still need to `return false` to terminate exection. :-(